rewrite-maven/src/main/java/org/openrewrite/maven/cache/RocksdbMavenPomCache.java

```diff
@@ -297,6 +300,31 @@ private void fillUnresolvablePoms() {
             } catch (RocksDBException exception) {
                 throw new IllegalStateException("Unable to create cache database." + exception.getMessage(), exception);
             }
+
+            try {
+                cleanCacheIfCorrupt(workspace);
+            } catch (Exception ex) {
+                throw new IllegalStateException("Unable to clear corrupt maven pom cache.", ex);
+            }
+        }
+
+        private void cleanCacheIfCorrupt(String workspace) throws IOException {
+            try {
+                database.verifyChecksum();
+            } catch (RocksDBException ex){
+                if (ex.getMessage().contains("block checksum mismatch")) {
+                    File workDirectory = new File(workspace);
```

Comment:
isn't `workspace` == `~/.rewrite/cache` shouldnt we only be deleting `~/.rewrite/cache/pom`?

Reply:
I'll update the naming so that it's clearer.
Currently, the RocksdbMavenCache constructor takes in a Path `workspace` and concatenates `.pom`. So, the user sets the workspace, but we add a .pom directory.
The pomCacheDir.getAbsolutePath() is what is passed around, so it is only deleting files in the .pom folder.

Reply:
oh, we are good here, i just needed to go back and look at the whole file.